File: C++/program65.cpp
#include<iostream>
using namespace std;

class Reverse
{
    private:
        int iNo;
    public:
        Reverse(int No)
        {
            if(No < 0)
            {
                No = -No;
            }

            iNo = No;

        }

        int DisplayReverse()
        {
            int iDigit = 0, iRev = 0;

            while(iNo != 0)
            {
                iDigit = iNo % 10;
                iRev = (iRev * 10) + iDigit;
                iNo = iNo / 10;
            }

            return iRev;
        }
};

int main()
{
    int iValue = 0, iRet = 0;

    cout<<"Enter number"<<'\n';
    cin>>iValue;

    Reverse robj(iValue);
    iRet = robj.DisplayReverse();

    cout<<"Reverse number is :  "<<iRet<<'\n';

    return 0;

}

File: Python/program65.py
class FcatorNonFactor:
    _No = 0
    
    def __init__(self, iNo):
        self._No = iNo
           
    def SumFactorNonFactor(self):
        sum1 = 0
        sum2 = 0
        for iCnt in range(1,self._No,1):
            if((self._No % iCnt) == 0): 
                sum1 = sum1 + iCnt
            else:
                sum2 = sum2 + iCnt
        return sum1,sum2
        
           
 
    
def main(): 
    
    iValue = int (input("Enter number : "))
        
    sobj = FcatorNonFactor(iValue)
    iRet = sobj.SumFactorNonFactor()
    
    print("Summation of factor and  NON factor is ",iRet)

if __name__ == "__main__":
    main()